From ee0aaeb90865a30bd11ec5539c14a4c0f5aa33e5 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Thu, 11 Apr 2024 16:01:49 +0200
Subject: [PATCH 1108/1141] ARM: dts: stm32: fix irq type of touchscreen on
 stm32mp157 EV1 boards

The touchscreen driver goodix_ts ignores the interrupt type set in
the DT; it queries the touchscreen firmware for the interrupt type
to be requested.
When the touchscreen device is unbound, the interrupt keeps the
type already requested.
On a following bind of the same driver, the interrupt framework
will find that the interrupt is already registered and will check
if its interrupt type matches the DT; in case of type mismatch the
bind will fail.

The touchscreen firmware can be update during the product life,
thus the interrupt type can change.

Use IRQ_TYPE_NONE in the DT, as it works as a wildcard and matches
any interrupt type requested by the firmware.

Current touchscreen firmware in stm32mp157 EV1 require falling
edge type, that mismatches the type in DT and prevents re-binding
the device.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I824c84bf0295feee111a66ff0409a9528ce2f8bd
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/373137
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 arch/arm/boot/dts/stm32mp157a-ev1.dts | 2 +-
 arch/arm/boot/dts/stm32mp157c-ev1.dts | 2 +-
 arch/arm/boot/dts/stm32mp157d-ev1.dts | 2 +-
 arch/arm/boot/dts/stm32mp157f-ev1.dts | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

--- a/arch/arm/boot/dts/stm32mp157a-ev1.dts
+++ b/arch/arm/boot/dts/stm32mp157a-ev1.dts
@@ -499,7 +499,7 @@
 		VDDIO-supply = <&v3v3>;
 		status = "okay";
 
-		interrupts = <14 IRQ_TYPE_EDGE_RISING>;
+		interrupts = <14 IRQ_TYPE_NONE>;
 		interrupt-parent = <&stmfx_pinctrl>;
 	};
 };
--- a/arch/arm/boot/dts/stm32mp157c-ev1.dts
+++ b/arch/arm/boot/dts/stm32mp157c-ev1.dts
@@ -501,7 +501,7 @@
 		VDDIO-supply = <&v3v3>;
 		status = "okay";
 
-		interrupts = <14 IRQ_TYPE_EDGE_RISING>;
+		interrupts = <14 IRQ_TYPE_NONE>;
 		interrupt-parent = <&stmfx_pinctrl>;
 	};
 };
--- a/arch/arm/boot/dts/stm32mp157d-ev1.dts
+++ b/arch/arm/boot/dts/stm32mp157d-ev1.dts
@@ -499,7 +499,7 @@
 		VDDIO-supply = <&v3v3>;
 		status = "okay";
 
-		interrupts = <14 IRQ_TYPE_EDGE_RISING>;
+		interrupts = <14 IRQ_TYPE_NONE>;
 		interrupt-parent = <&stmfx_pinctrl>;
 	};
 };
--- a/arch/arm/boot/dts/stm32mp157f-ev1.dts
+++ b/arch/arm/boot/dts/stm32mp157f-ev1.dts
@@ -502,7 +502,7 @@
 		VDDIO-supply = <&v3v3>;
 		status = "okay";
 
-		interrupts = <14 IRQ_TYPE_EDGE_RISING>;
+		interrupts = <14 IRQ_TYPE_NONE>;
 		interrupt-parent = <&stmfx_pinctrl>;
 	};
 };
